test/models/audio_version_template_test.rb

```diff
@@ -74,4 +74,21 @@
     audio_version_template.validate_audio_file(audio_file).must_include 'is not in video format'
   end
 
+  it 'updates segment count when child file templates change' do
+    audio_version_template.audio_file_templates.count.must_equal 0
+    audio_version_template.segment_count.must_be_nil
+
+    f1 = audio_version_template.audio_file_templates.create!(label: 'seg1')
+    f1.run_callbacks(:commit)
```

Comment:
I seem to recall this is how we handle these in tests, though I applaud your efforts to get rid of needing to invoke.